stack.yaml

```diff
@@ -2,6 +2,7 @@ extra-deps:
 - fused-effects-lens-1.2.0.1@sha256:675fddf183215b6f3c1f2a0823359a648756435fd1966284e61830ec28ad61fa,1466
 - hsnoise-0.0.3@sha256:260b39175b8a3e3b1719ad3987b7d72a3fd7a0fa99be8639b91cf4dc3f1c8796,1476
 - simple-enumeration-0.2.1@sha256:8625b269c1650d3dd0e3887351c153049f4369853e0d525219e07480ea004b9f,1178
+- servant-docs-0.12@sha256:73fd8901a9dbe4d2c563b7da0b94c75f7755ee9059eec80ff621f415a8d3ef6a,3383
```

Comment:
Note that the specific revision of `servant-docs` identified here by hash has an upper bound of `base < 4.17` so it won't build with GHC 9.4 (which comes with our selected resolver).  However, the most recent revision of `servant-docs` on Hackage does work.  I was able to build simply by removing the `@sha256` part; I am not sure how to get `stack` to tell me the right hash to replace it with.

Reply:
A couple of things:
1. I'm not sure what's up with the `base` upper bound, but it builds locally for me with `stack` already:
    ```
    ~/github/swarm$ stack ghc -- --version
    The Glorious Glasgow Haskell Compilation System, version 9.4.4
    ~/github/swarm$ stack build --fast
    ~/github/swarm$ echo $?
    0
    ```
2. Our Cabal-based CI builds are all passing, but if there is an additional merge precondition of successful `stack` builds, we should add `stack` to CI.

Reply:
Weird.  This is what I get:
```
diophantus :: ~/projects/swarm » stack --version
Version 2.9.3, Git revision 6cf638947a863f49857f9cfbf72a38a48b183e7e x86_64 hpack-0.35.1
diophantus :: ~/projects/swarm » stack clean --full
diophantus :: ~/projects/swarm » stack ghc -- --version
The Glorious Glasgow Haskell Compilation System, version 9.4.4
diophantus :: ~/projects/swarm » stack build --fast

Error: [S-4804]
       Stack failed to construct a build plan.
       
       While constructing the build plan, Stack encountered the following errors:
       
       In the dependencies for servant-docs-0.12:
           base-4.17.0.0 from Stack configuration does not match >=4.9 && <4.17 (latest matching version is 4.16.4.0)
           lens-5.2 from Stack configuration does not match >=4.17 && <5.2 (latest matching version is 5.1.1)
       needed due to swarm-0.3.0.1 -> servant-docs-0.12
       
       Some different approaches to resolving this:
       
         * Set 'allow-newer: true' in /home/brent/.stack/config.yaml to ignore all version constraints and build anyway.
       
         * Build requires unattainable version of base. Since base is a part of GHC, you most likely need to use a different GHC version with the matching
           base.    
```

> 2\. if there is an additional merge precondition of successful `stack` builds, we should add `stack` to CI.

Yes, you're right.  I think the intention has always been to support building both with `cabal` and with `stack`, so our CI should test that.  I'll look into it.